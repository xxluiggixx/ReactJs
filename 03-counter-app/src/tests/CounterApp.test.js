const { shallow } = require("enzyme");
const { default: CounterApp } = require("../CounterApp");
import "@testing-library/jest-dom";
import React from "react";
describe("Pruebas de CounterApp", () => {
  /*#tarea 
    1. Crear las siguientes pruebas para el <CounterApp/>
        *debe de mostrar <CounterApp/> correctamente (hacer match con un snapshot)
            y sus valores por defecto
        
        *debe de mostrar el valor por defecto de 100
            *usar el wrapper.find, tomando el elemento html donde se muestra el valor del contador
    */
  let wrapper = shallow(<CounterApp />);
  beforeEach(() => {
    wrapper = shallow(<CounterApp />);
  });
  test("Debe mostrar <CounterApp/> correctamente ", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("Debe de mostrar el valor por defecto de 100", () => {
    const wrapper = shallow(<CounterApp value={100} />);
    const counterH2 = wrapper.find("h2").text().trim();
    expect(counterH2).toBe("100");
  });
  test("Debe de incrementar con el botón +1", () => {
    const btnPlus = wrapper.find("button").at(0).simulate("click");
    const counterH2 = wrapper.find("h2").text().trim();
    expect(counterH2).toBe("101");
  });
  test("Debe de establecer el valor por defecto con el botón Reset", () => {
    const wrapper = shallow(<CounterApp value={105} />);
    const btnPlus = wrapper.find("button").at(2).simulate("click");
    const btnReset = wrapper.find("button").at(1).simulate("click");
    const counterH2 = wrapper.find("h2").text().trim();
    expect(counterH2).toBe("105");
  });
  test("Debe de decrementar con el botón -1", () => {
    const btnPlus = wrapper.find("button").at(2).simulate("click");
    const counterH2 = wrapper.find("h2").text().trim();
    expect(counterH2).toBe("99");
  });
});
