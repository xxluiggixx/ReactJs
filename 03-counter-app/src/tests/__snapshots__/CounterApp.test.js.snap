// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de CounterApp Debe mostrar <CounterApp/> correctamente  1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    100
  </h2>
  <button
    id="+1"
    onClick={[Function]}
  >
    +1
  </button>
  <button
    id="Reset"
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    id="-1"
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
